src/runtime/virtcontainers/utils/utils_linux.go

```diff
@@ -151,3 +153,59 @@ func IsAPVFIOMediatedDevice(sysfsdev string) bool {
 	}
 	return false
 }
+
+func waitProcessUsingPidfd(pid int, timeoutSecs uint, logger *logrus.Entry) (bool, error) {
+	pidfd, err := unix.PidfdOpen(pid, 0)
+
+	if err != nil {
+		if err == unix.ESRCH {
+			return false, nil
+		}
+
+		return true, err
+	}
+
+	defer unix.Close(pidfd)
+
+	start := time.Now()
+	var n int
+
+	for {
+		remaining := int(timeoutSecs*1000) - int(time.Since(start).Milliseconds())
+		if remaining < 0 {
+			remaining = 0
+		}
+
+		n, err = unix.Poll([]unix.PollFd{{Fd: int32(pidfd), Events: unix.POLLIN}}, remaining)
+		if err != unix.EINTR {
+			break
+		}
+	}
+
+	if err != nil || n != 1 {
```

Comment:
Did you observe the case `n != 1`? What is it for?

To me, the underlying syscall could get you -1, 0 or 1. The case where you'd get 0, you'd stay in the loop because of the `EINTR` test. The case where you'd get -1, presumably `err` would be set. So I think the test is defensive coding. Not necessarily a bad thing, but might warrant a comment.